examples/components/TodoTextInput.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component, PropTypes } from 'react';
+
+export default class TodoTextInput extends Component {
+  static propTypes = {
+    onSave: PropTypes.func.isRequired
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      text: this.props.text || ''
+    };
+  }
+
+  componentDidUpdate() {
+    // Sets the cursor position at the end of todo.
+    // edit variable is used to keep focus on new-todo otherwise new-todo loses
+    // focus after entering a todo.
+    if (this.props.class === 'edit' && this.props.edit) {
```

Comment:
Do you mean `className`?
I think this is a bad practice. Its parent could pass an `edit` property to handle this case.